### ISSUE TITLE
Suggestions to user regarding current active tool in the bottom infobar.

### ISSUE THREAD

**Issue by keshav2010** [bjorn/tiled#1855]
While working with Tiled, I noticed that the bottom most info-bar (which often displays information such as mouse coordinates along with the grid's coordinates we are currently pointing to) can be utilized to display information related to tool which is at that instant of time being used by user, specially for tools such as 
"Insert Polygon" where i believe it might be a good step to display information such as which key to press to confirm the polygon constructed so far or key which can be pressed to remove it.

![capture](https://user-images.githubusercontent.com/531764/34907479-56b5ec5a-f87f-11e7-8d89-5df9165e40fb.PNG)

**Comment by bjorn** [comment]
Yeah, the status bar could definitely be more helpful. I think Inkscape sets a good example for this.

**Comment by keshav2010** [comment]
yeah, i will be looking over to implement this feature, as of now, im wondering if its a good choice to implement this as a tiny system in itself, that will be notified about current active tool and fetch information related to it, translate it into user's choice of language (English, etc) and display it on status bar ?
i will be however having a look over the code, since my current understanding regarding how Tiled's way of interacting with this status bar is vague, to see how information is currently being passed on to the status bar (regarding mouse position and grid location) , although its numerical data so no translation would be required, this might eventually help me figure out a proper implementation, if there is any suggestion on this issue that would be very helpful.

**Comment by satu0king** [comment]
I would love to work on this feature. @bjorn any tips on how to implement this?

**Comment by keshav2010** [comment]
@satu0king I'm still figuring out a proper way to implement this, since this feature should be implemented as such so as to make sure if any new tool is added into Tiled in future, it should be fairly easy to link its information. 
I'm still not sure which way to follow, a separate class that maps information related to a tool by treating its name as Key or we can go the other way around and just alter the current existing Abstract-tool class. 
Would be glad to have your participation in this too😃

**Comment by satu0king** [comment]
Cool loop me in when u start working on it. I think having an XML file which has entries for each tool item should work fine. When the user clicks on the tool, the status bar can be updated.

**Comment by bjorn** [comment]
It could be done either as part of or similar to `AbstractTool::statusInfo`.

**Comment by keshav2010** [comment]
Alright, i will begin implementing it for tools that can be selected while working on an Object Layer (such as insert-polygon, polyline, etc). 
Thanks for the suggestion.
